ext/src/ruby_api/memory.rs

```diff
@@ -131,19 +131,32 @@ impl<'a> Memory<'a> {
     /// Read +size+ bytes starting at +offset+ into a {Slice}. This
     /// provides a way to read a slice of memory without copying the data.
     ///
-    /// Resizing the memory (as with `Wasmtime::Memory#grow`) will invalidate
+    /// The returned {Slice} lazily reads the underlying memory, meaning that
+    /// the actual pointer to the string buffer is not materialzed until
+    /// `Slice#to_str` is called.
+    ///
+    /// SAFETY: Resizing the memory (as with `Wasmtime::Memory#grow`) will invalidate
     /// the slice, causing any attempt to read the slice to raise an error.
+    /// However, it is not possible to invalidate the Ruby {String} object after
+    /// calling `Slice#to_str`.
+    ///
+    /// As such, the caller must ensure that the Wasmtime {Memory} is not
+    /// resized will the while holding the Ruby string.  Failing to do so could
```

Comment:
```suggestion
    /// resized while holding the Ruby string.  Failing to do so could
```

I think this is what you meant?